src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -173,7 +161,8 @@ DLLEXPORT ICorJitCompiler* getJit()
 
     if (ILJitter == nullptr)
     {
-        ILJitter = new (CILJitSingleton) CILJit();
+        alignas(alignof(CILJit)) static char CILJitBuff[sizeof(CILJit)];
```

Comment:
Actually, is there a reason why we cannot just use a regular `static CILJit` singleton and just return its address here?

Reply:
https://github.com/dotnet/runtime/pull/73471

Reply:
I was not sure if there were concerns with the potential static constructor or ordering with `jitStartup` (there is a check that initialization has run before we construct this).

Reply:
`CILJit` constructor does close to nothing, it is just initializing the vtable. So the ordering is not a concern.